htdocs/discover_tab.js

```diff
@@ -0,0 +1,58 @@
+var discover = function () {
+
+    /* "Model-Model-View-Controller"
+     .. or something ..
+     there are two models: the local state and the global rcs+github
+     the local model is private data of this class
+    */
+
+    // local model (all caches of stuff stored in RCS/github)
+    var notebook_info_ = {}, // all notebooks we are aware of
+        num_stars_ = {}, // number of stars for all known notebooks
+        fork_count_ = {};
+
+    function clean_r(obj) { 
+        delete obj.r_attributes; 
+        delete obj.r_type; 
+        return obj; 
+    };
+
+    function load_everything(){
+        var allnotebooks = rcloud.config.get_all_notebook_info();
```

Comment:
This should be just the list of notebooks that are actually shown on the discovery page; loading the metadata for all known notebooks is prohibitively slow.